core/src/test/java/org/elasticsearch/VersionTests.java

```diff
@@ -230,7 +232,7 @@ public void testParseVersion() {
         });
         assertSame(Version.CURRENT, Version.fromString(Version.CURRENT.toString()));
 
-        assertSame(Version.fromString("2.0.0-SNAPSHOT"), Version.fromString("2.0.0"));
+        assertEquals(Version.fromString("2.0.0-SNAPSHOT"), Version.fromId(2000099));
```

Comment:
did you have issues with fromString or was it to be consistent with other places where you used fromId?